### PR TITLE
2019 03 emit blush

### PR DESCRIPTION
**Pull request by cordellcalitz**
* allow auth token in URL
 * Emit _msearch events

### CODE REVIEW

File: lib/components/proxy/rules-engine.js [hunk 1]
```diff
@@ -98,13 +103,53 @@ RulesEngine.prototype.enforce = function(stackname, facts, consequenceHandler){
...
+                  if (_.has(query.query_string, "query") === false) return;
+                  query_paths.push(query.query_string.query);
+                });
+              });
+              if (query_paths.length > 0) {
+                var emitData = {
+                  query_indexes: query_indexes,
+                  query_paths: query_paths
+                };
+                console.log("emit msearch-path-request", emitData);
```
Comment (JensEggers) [inline lib/components/proxy/rules-engine.js]: rather use happner debug log

File: lib/components/proxy/rules-engine.js [hunk 2]
```diff
@@ -98,13 +103,53 @@ RulesEngine.prototype.enforce = function(stackname, facts, consequenceHandler){
...
+                _this.$happn.emit("msearch-path-request", emitData);
+              }
+              resolve(facts);
+            });
+          } catch (e) {
+            console.log("Error emitting blush event", e);
```
Comment (JensEggers) [inline lib/components/proxy/rules-engine.js]: rather use happner error log

File: lib/components/proxy/rules-engine.js (hunk 1 above)
Comment (cordellcalitz) [reply lib/components/proxy/rules-engine.js]: done

File: lib/components/proxy/rules-engine.js (hunk 2 above)
Comment (cordellcalitz) [reply lib/components/proxy/rules-engine.js]: done

File: lib/components/proxy/rules/test.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (JensEggers) [inline lib/components/proxy/rules/test.json]: is this file being used? should maybe go into test folder

Comment (cordellcalitz) [reply lib/components/proxy/rules/test.json]: I actually think this was an accidental commit and we can remove it. Will double check with JW

Comment (cordellcalitz) [reply lib/components/proxy/rules/test.json]: Removed

**Pull request merged**